src/pybind/mgr/dashboard_v2/tools.py

```diff
@@ -347,3 +349,89 @@ def _callback(self):
             if name in cookie:
                 del cookie[name]['expires']
                 del cookie[name]['max-age']
+
+
+class NotificationQueue(threading.Thread):
+    _ALL_TYPES_ = '__ALL__'
+    _listeners = collections.defaultdict(lambda: [])
+    _lock = threading.Lock()
+    _cond = threading.Condition()
+    _queue = collections.deque()
+    _running = False
+
```

Comment:
I'd initialize `cls._instance = None` here and add this to `stop()`
```python
if cls._instance is not None: cls._instance.join()
```
Otherwise this will crash, if `stop()` is called before `start_queue()`